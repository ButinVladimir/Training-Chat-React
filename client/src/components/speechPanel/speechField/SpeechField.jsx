import React from 'react';
import PropTypes from 'prop-types';
import ToButton from './ToButtonContainer';
import './style.css';

function SpeechField({
  speech,
  to,
  onChange,
  onSay,
}) {
  const convertedTo = to.map(t => <ToButton key={t} to={t} />);

  return (
    <div className="speech-container">
      <div className="speech-input-container">
        <input className="speech" type="text" value={speech} onChange={e => onChange(e.target.value)} />
        <button className="say" type="button" onClick={onSay}>Say</button>
      </div>
      <div className="speech-to-container">
        {convertedTo}
      </div>
    </div>
  );
}

SpeechField.propTypes = {
  speech: PropTypes.string,
  to: PropTypes.arrayOf(PropTypes.string).isRequired,
  onChange: PropTypes.func.isRequired,
  onSay: PropTypes.func.isRequired,
};

SpeechField.defaultProps = {
  speech: '',
};

export default SpeechField;
