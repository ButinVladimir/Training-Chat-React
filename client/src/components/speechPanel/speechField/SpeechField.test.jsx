import React from 'react';
import { shallow } from 'enzyme';
import SpeechField from './SpeechField';

describe('SpeechField', () => {
  it('renders without crashing', () => {
    const component = shallow(
      <SpeechField
        speech="123"
        to={[]}
        onChange={() => {}}
        onSay={() => {}}
      />,
    );
    expect(component).toMatchSnapshot();
  });

  it('renders without crashing when message is being sent to 1 recipient', () => {
    const component = shallow(
      <SpeechField
        speech="123"
        to={['recipient1']}
        onChange={() => {}}
        onSay={() => {}}
      />,
    );
    expect(component).toMatchSnapshot();
  });

  it('renders without crashing when message is being sent to 2 recipients', () => {
    const component = shallow(
      <SpeechField
        speech="123"
        to={['recipient1', 'recipient2']}
        onChange={() => {}}
        onSay={() => {}}
      />,
    );
    expect(component).toMatchSnapshot();
  });

  it('renders without crashing when message is being sent to 3 recipients', () => {
    const component = shallow(
      <SpeechField
        speech="123"
        to={['recipient1', 'recipient2', 'recipient3']}
        onChange={() => {}}
        onSay={() => {}}
      />,
    );
    expect(component).toMatchSnapshot();
  });

  it('handles speech change', () => {
    const mock = jest.fn();
    const component = shallow(
      <SpeechField
        speech="123"
        to={[]}
        onChange={mock}
        onSay={() => {}}
      />,
    );

    const input = component.find('input.speech');
    input.value = 'aaa';
    input.simulate('change', { target: { value: 'value' } });

    expect(mock).toHaveBeenCalledTimes(1);
    expect(mock).toHaveBeenCalledWith('value');
  });

  it('handles saying', () => {
    const mock = jest.fn();
    const component = shallow(
      <SpeechField
        speech="123"
        to={[]}
        onChange={() => {}}
        onSay={mock}
      />,
    );
    const input = component.find('button.say');
    input.simulate('click');

    expect(mock).toHaveBeenCalledTimes(1);
  });
});
