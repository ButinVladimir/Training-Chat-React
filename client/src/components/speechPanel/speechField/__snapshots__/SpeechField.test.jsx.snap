// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpeechField renders without crashing 1`] = `
<div
  className="speech-container"
>
  <div
    className="speech-input-container"
  >
    <input
      className="speech"
      onChange={[Function]}
      type="text"
      value="123"
    />
    <button
      className="say"
      onClick={[Function]}
      type="button"
    >
      Say
    </button>
  </div>
  <div
    className="speech-to-container"
  />
</div>
`;

exports[`SpeechField renders without crashing when message is being sent to 1 recipient 1`] = `
<div
  className="speech-container"
>
  <div
    className="speech-input-container"
  >
    <input
      className="speech"
      onChange={[Function]}
      type="text"
      value="123"
    />
    <button
      className="say"
      onClick={[Function]}
      type="button"
    >
      Say
    </button>
  </div>
  <div
    className="speech-to-container"
  >
    <ToButton
      key="recipient1"
      onRemoveTo={[Function]}
      to="recipient1"
    />
  </div>
</div>
`;

exports[`SpeechField renders without crashing when message is being sent to 2 recipients 1`] = `
<div
  className="speech-container"
>
  <div
    className="speech-input-container"
  >
    <input
      className="speech"
      onChange={[Function]}
      type="text"
      value="123"
    />
    <button
      className="say"
      onClick={[Function]}
      type="button"
    >
      Say
    </button>
  </div>
  <div
    className="speech-to-container"
  >
    <ToButton
      key="recipient1"
      onRemoveTo={[Function]}
      to="recipient1"
    />
    <ToButton
      key="recipient2"
      onRemoveTo={[Function]}
      to="recipient2"
    />
  </div>
</div>
`;

exports[`SpeechField renders without crashing when message is being sent to 3 recipients 1`] = `
<div
  className="speech-container"
>
  <div
    className="speech-input-container"
  >
    <input
      className="speech"
      onChange={[Function]}
      type="text"
      value="123"
    />
    <button
      className="say"
      onClick={[Function]}
      type="button"
    >
      Say
    </button>
  </div>
  <div
    className="speech-to-container"
  >
    <ToButton
      key="recipient1"
      onRemoveTo={[Function]}
      to="recipient1"
    />
    <ToButton
      key="recipient2"
      onRemoveTo={[Function]}
      to="recipient2"
    />
    <ToButton
      key="recipient3"
      onRemoveTo={[Function]}
      to="recipient3"
    />
  </div>
</div>
`;
